IntelliTect.TestTools.Selenate/IntelliTect.TestTools.Selenate.Tests/BaseTest.cs

```diff
@@ -0,0 +1,76 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Text;
+
+namespace IntelliTect.TestTools.Selenate.Tests
+{
+    public class BaseTest
+    {
+        protected void TestVoidException()
+        {
+            throw new Exception();
+        }
+
+        protected bool TestReturnException()
+        {
+            throw new Exception();
+        }
+
+        protected void TestVoidNullRefException()
+        {
+            throw new NullReferenceException();
+        }
+
+        protected void TestReturnNullRefException()
+        {
+            throw new NullReferenceException();
+        }
+
+        protected void CheckExceptionsVoidReturn(int secondsToFail = 1, int numberOfDifferentExceptions = 1)
+        {
+            ThrowExceptions(secondsToFail, numberOfDifferentExceptions);
+        }
+
+        protected bool CheckExceptionsBoolReturn(int secondsToFail = 1, int numberOfDifferentExceptions = 1)
+        {
+            ThrowExceptions(secondsToFail, numberOfDifferentExceptions);
+            return true;
+        }
+
+        private void ThrowExceptions(int secondsToFail, int numberOfExceptions)
+        {
+            if (_Timeout == TimeSpan.MinValue)
+            {
+                _Timeout = TimeSpan.FromSeconds(secondsToFail);
```

Comment:
For use inside of this method I would have a local variable `timeOut` rather than setting the field. By using the field you can run into problems if the tests are run in parallel.

Reply:
I'll leave the method as-is for now (XUnit reconstructs for every test, so that should avoid issues at least for now,) but I'll follow-up with you when I get a chance. Basically I just need a different (but known) exception thrown every time this method is invoked up until the timeout is reached.